python/lsst/cp/verify/verifyStats.py

```diff
@@ -0,0 +1,510 @@
+            dictionaries of the statistics measured and their values.
+        """
+
+        outputStats = self._emptyAmpDict(exposure)
+
+        if len(self.config.imageStatKeywords):
+            outputStats = mergeStatDict(outputStats, self.amplifierStats(exposure,
+                                                                         self.config.imageStatKeywords,
+                                                                         statControl))
+
+        if len(self.config.unmaskedImageStatKeywords):
+            outputStats = mergeStatDict(outputStats, self.unmaskedImageStats(exposure, statControl))
+
+        if len(self.config.normImageStatKeywords):
+            outputStats = mergeStatDict(outputStats, self.normalizedImageStats(exposure, statControl))
+
+        if len(self.config.crImageStatKeywords):
+            outputStats = mergeStatDict(outputStats, self.crImageStats(exposure, statControl))
+
+        return outputStats
+
+    def amplifierStats(self, exposure, keywordDict, statsControl):
+        """Measure amplifier level statistics on the data read from disk.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            The exposure to measure.
+        keywordDict : `dict` [`str`, `str`]
+            A dictionary of keys to use in the output results, with
+            values the string name associated with the
+            `lsst.afw.math.statistics.Property` to measure.
+        statControl : `lsst.afw.math.StatControl`
+            Statistics control object with the parameters defined by
+            the config.
+
+        Returns
+        -------
+        ampStats : `dict` [`str`, `dict` [`str`, scalar]]
+            A dictionary indexed by the amplifier name, containing
+            dictionaries of the statistics measured and their values.
+        """
+
+        ampStats = {}
+
+        # Convert the string names in the keywordDict to the afwMath values.
+        # The statisticToRun is then the bitwise-or of that set.
+        statisticToRun = 0
+        statAccessor = {}
+        for k, v in keywordDict.items():
+            statValue = afwMath.stringToStatisticsProperty(v)
+            statisticToRun |= statValue
+            statAccessor[k] = statValue
+
+        # Measure stats on all amplifiers.
+        for ampIdx, amp in enumerate(exposure.getDetector()):
+            ampName = amp.getName()
+            theseStats = {}
+            ampExp = exposure.Factory(exposure, amp.getBBox())
+            stats = afwMath.makeStatistics(ampExp.getMaskedImage(), statisticToRun, statsControl)
+
+            for k, v in statAccessor.items():
+                theseStats[k] = stats.getValue(v)
+            ampStats[ampName] = theseStats
+
+        return ampStats
+
+    def unmaskedImageStats(self, exposure, statControl):
+        """Measure amplifier level statistics on the data read from disk,
+        ignoring all mask planes.
```

Comment:
I think it's often (incorrectly) assumed that the "ignoring all mask planes" is to be interpreted as "ignoring all pixels with any mask plane set".  I'm definitely not insisting, but it may be worthwhile to rephrase along the lines of "including all pixels in the exposure, regardless of any mask planes set".